### PR TITLE
sample id selection

### PR DESCRIPTION
**Pull request by paulocilasjr**
Confirm sample-ID handling for leakage-aware splits (grouped train/test split only when no test file; group-aware CV via GroupKFold when sample IDs are provided) and ensure graceful fallback when sample IDs are invalid or missing.

### CODE REVIEW

File: tools/tabularlearner/base_model_trainer.py [hunk 1]
```diff
@@ -165,6 +188,100 @@ def load_data(self):
...
+                for group_id in group_ids:
+                    split_idx = _choose_split(counts, targets, active)
+                    counts[split_idx] += len(group_to_indices[group_id])
+                    if split_idx == 0:
+                        train_idx.extend(group_to_indices[group_id])
+                    else:
+                        test_idx.extend(group_to_indices[group_id])
+
+                self.test_data = self.data.loc[test_idx].reset_index(drop=True)
+                self.data = self.data.loc[train_idx].reset_index(drop=True)
```
Comment (qchiujunhao) [inline tools/tabularlearner/base_model_trainer.py]: Thank you for the great changes @paulocilasjr ! This could be future work: there’s a boundary case to watch for: with very few or highly imbalanced groups, it’s possible to end up with an empty train or test set. The current code doesn’t guard against this, which could cause PyCaret to error or produce invalid evaluation. It would be safer to check len(train_idx)==0 or len(test_idx)==0 and either back to a non-group split or surface a clear warning.

**Review by qchiujunhao — APPROVED** [review]
Thank you @paulocilasjr
